### DISCUSSION TITLE
Email notification, extension update, in bulk

### DISCUSSION THREAD

**Discussion by komir2004** [akeeba/panopticon#298] [Ideas]
Hello, I have been trying to set up the email notification feature, but I have not succeeded yet.
According to the documentation, it seems that Panopticon sends a separate email for each web page that has an extension update.
I think it would be better to receive one email with all the updates, and have the extensions grouped by the websites they belong to.
Example of email

Web site 1
-extension 1
-extension 2

Web site 2
-extension 1
-extension 2

**Comment by nikosdion** [comment]
As already explained, this is not something which will be implemented.

Y'all don't really "get" the reason for having instant update notifications by email as opposed to periodically visiting Panopticon to install them. Let's try to explain that.

Updates are not discovered by Panopticon, they are discovered by Joomla. Joomla caches updates for 6 to 24 hours. Each extension may be at a different stage in its caching lifetime, i.e. Joomla! does not aways fetch update information for all extensions at the exact same time. This means that you will not have a complete picture of the updates available on a site for anywhere between 12 to 48 hours.

If you have multiple sites in a group, you will not have a complete picture of their updates for twice that long, i.e. 24 to 96 hours (1 to 4 days).

You are basically asking to receive an email every week with the updates. This is pointless. The whole point of the email is to be notified as soon as your site knows there is an update available so you can install it. This is very important if there are critical security updates. 

The average time between releasing a new version with a security fix and someone coming up with exploitation code is **8 hours**. This means that if you don't install security updates as soon as they are released chances are your site will be hacked (always depending on the severity of the security issue which was addressed). That's the only reason you need these emails and that's why the emails need to be as instantaneous as possible.

Therefore, obliterating any utility of the ahead warning feature I have given you is not something I am interested in doing.

If you don't care about installing updates immediately, you don't need an email about them either. You just need to log into Panopticon once a week and install updates (which will be easier with the new Extension Updates Overview page coming in 1.0.4).

**Comment by nikosdion** [comment]
I thought about it some more. I can see some use cases where it _does_ make sense to have scheduled summaries. See my notes in https://github.com/akeeba/panopticon/issues/301 on the subject. Please let me know if this kind of implementation would work for your use case.
